sdk/identity/identity/src/credentials/managedIdentityCredential/utils.ts

```diff
@@ -42,6 +42,7 @@ export interface TokenResponseParsedBody {
   expires_in: number;
   expires_on?: number | string;
   refresh_in?: number;
+  refresh_on?: number | string;
```

Comment:
huh weird to see both `refresh_in` and `refresh_on` - do you know why we need both? Do we need both?

Reply:
refresh_in was there already as part of managedIdentity. refresh_on I added now, these are internal ones though to help us read from msal

Reply:
References - https://github.com/Azure/azure-sdk-for-js/pull/25213 and https://github.com/Azure/azure-sdk-for-js/pull/25535. I probably should clean up and remove refreshIn since it was added as proactive token refresh logic, which was then replaced by msal implementation.

Reply:
Got it. I don't have a full understanding of these butttt if we can get away with cleaning up refreshIn separately then it won't need to block your PR - do you think it makes sense to create an issue to tackle that separately or did you wanna do it as part of this PR? Up to you!

Reply:
No I just removed it here :)

Reply:
Ah lovely